rho/authaddcommand.py

```diff
@@ -15,28 +15,50 @@
 """
 
 from __future__ import print_function
-import csv
+import json
 import os
 import sys
 import uuid
 from getpass import getpass
 from collections import OrderedDict
 from rho.clicommand import CliCommand
+from rho.vault import read_vault_password, get_vault_password, Vault
 from rho.translation import get_translation
 
 _ = get_translation()
 
 
-# Write new credentials in the related file.
-def _save_cred(cred):
-    append_or_write = 'a'
+def auth_exists(cred_list, auth_name):
+    """ Checks whether authentication credential already exists
+    :param cred_list: A list of credential dictionaries
+    :param auth_name: Name of credential to check for existence
+    :returns: True if auth_name exists, False otherwise
+    """
+    auth_found = False
+    for cred in cred_list:
+        if cred.get('name') == auth_name:
+            auth_found = True
+            break
+    return auth_found
+
+
+def _save_cred(vault, new_cred, cred_list):
+    """ Write new credentials in the related file
+    :param vault: Vault object for writing encrypted data
+    :param new_cred: New credential to be added to the credentials file
+    :param cred_list: A list of existing credential dictionaries from the
+    credential file
+    """
+    credentials_path = 'data/credentials'
+
     if not os.path.exists('data'):
         os.makedirs('data')
-    if not os.path.isfile('data/credentials'):
-        append_or_write = 'w'
-    with open('data/credentials', append_or_write) as cred_file:
-        dict_writer = csv.DictWriter(cred_file, cred.keys())
-        dict_writer.writerow(cred)
+
+    cred_list.append(new_cred)
+
+    data = json.dumps(cred_list, separators=(',', ': '))
+    with open(credentials_path, 'wb') as cred_file:
```

Comment:
moved writing data into vault file. It now takes an object then coverts to json, the writes to temporary file, the moves to final file location.